.bazelversion

```diff
@@ -1 +1 @@
-3.7.2
+4.2.1
```

Comment:
Any side effect about diverging version from https://github.com/tensorflow/tensorflow/blob/master/.bazelversion ?

Reply:
Potentially, and if so this might justify a new approach to handle macOS TFA builds perhaps with a `macos_m1` branch. 

Created an issue to continue the discussion here: https://github.com/tensorflow/addons/issues/2616

Reply:
If we don't have a new rollback we are at 4.2.1 in https://github.com/tensorflow/tensorflow/blob/master/.bazelversion